snappy/overlord.go

```diff
@@ -0,0 +1,89 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package snappy
+
+import (
+	"path/filepath"
+
+	"github.com/ubuntu-core/snappy/dirs"
+	"github.com/ubuntu-core/snappy/progress"
+)
+
+// Overlord is responsible for the overall system state
+type Overlord struct {
+}
+
+// Install installs the given snap file to the systems.
+//
+// It returns the local snap file or an error
+func (o *Overlord) Install(snapFilePath string, origin string, inter progress.Meter, flags InstallFlags) (*SnapPart, error) {
+	return nil, ErrNotImplemented
+}
+
+// Uninstall remove the given local snap from the system
+//
+// It returns an error on failure
+func (o *Overlord) Uninstall(sp *SnapPart, pb progress.Meter) error {
+	return ErrNotImplemented
+}
+
+// SetActive sets the active state of the given snap
+//
+// It returns an error on failure
+func (o *Overlord) SetActive(sp *SnapPart, active bool, pb progress.Meter) error {
+	return ErrNotImplemented
+}
+
+// Installed returns the installed snaps from this repository
+func (o *Overlord) Installed() (parts []*SnapPart) {
+	globExpr := filepath.Join(dirs.SnapSnapsDir, "*", "*", "meta", "package.yaml")
+	if newParts, err := o.partsForGlobExpr(globExpr); err == nil {
```

Comment:
Thanks, this is a good question. 

It is easy to add,I would love to be able to do `FindSnapByname(overlord.Installed())` directly without the additional tedious:
```
installed, err := overlord.Installed()
if err != nil {
 return fmt.Errorf("somewhere something went wrong: %s", err)
}
FindSnapByName(installed)
```
and have Installed() just return the stuff that is correctly installed (might be empty on a ubuntu-desktop system that does not have anything yet). And if we need to find "broken" snaps or error conditions in the system maybe have Overlord.Error() or Overlord.State() or Overlord.FindBrokenSnaps() or similar? But I'm open to design suggestions here of course.